lib/Backend/GlobOpt.cpp

```diff
@@ -5166,9 +5174,22 @@ GlobOpt::OptInstr(IR::Instr *&instr, bool* isInstrRemoved)
 
     if (instr->HasBailOutInfo() && !this->IsLoopPrePass())
     {
-        this->currentBlock->globOptData.changedSyms->ClearAll();
-        this->currentBlock->globOptData.capturedValues = 
-            this->currentBlock->globOptData.capturedValuesCandidate;
+        GlobOptBlockData * globOptData = &this->currentBlock->globOptData;
+        globOptData->changedSyms->ClearAll();
```

Comment:
Also add comment here explaining what is going on.  The fact that the instr with the bailout might not have a bailout anymore, but if it does you need to use the changedSyms state to the point right after the CaptureValues call.